RecoLocalCalo/EcalRecAlgos/src/KernelHelpers.cu

```diff
@@ -59,18 +225,157 @@ namespace ecal {
+              if (near && top)
+                iquad = 1;
+              if (far && top)
+                iquad = 2;
+              if (far && bot)
+                iquad = 3;
+              if (near && bot)
+                iquad = 4;
+              
+              return iquad;
+            }
+            
+            __device__ int sector(int iX, int iY) {
+              //  Y (towards the surface)
+              //  T
+              //  |
+              //  |
+              //  |
+              //  o---------| X  (towards center of LHC)
+              //
+              static const int idx_[] = {
+                // 1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20
+                0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 9,
+                9, 9, 0, 0, 0, 0, 0, 0, 0,  // 20
+                0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 9,
+                9, 9, 9, 9, 9, 0, 0, 0, 0,  // 19
+                0, 0, 0, 2, 1, 1, 1, 1, 1, 1, 9,
+                9, 9, 9, 9, 9, 8, 0, 0, 0,  // 18
+                0, 0, 2, 2, 2, 1, 1, 1, 1, 1, 9,
+                9, 9, 9, 9, 8, 8, 8, 0, 0,  // 17
+                0, 2, 2, 2, 2, 1, 1, 1, 1, 1, 9,
+                9, 9, 9, 9, 8, 8, 8, 8, 0,  // 16
+                0, 2, 2, 2, 2, 2, 1, 1, 1, 1, 9,
+                9, 9, 9, 8, 8, 8, 8, 8, 0,  // 15
+                0, 2, 2, 2, 2, 2, 2, 1, 1, 1, 9,
+                9, 9, 8, 8, 8, 8, 8, 8, 0,  // 14
+                2, 2, 2, 2, 2, 2, 2, 2, 1, 1, 9,
+                9, 8, 8, 8, 8, 8, 8, 8, 8,  // 13
+                3, 3, 2, 2, 2, 2, 2, 2, 2, 0, 0,
+                8, 8, 8, 8, 8, 8, 8, 7, 7,  // 12
+                3, 3, 3, 3, 3, 3, 3, 2, 0, 0, 0,
+                0, 8, 7, 7, 7, 7, 7, 7, 7,  // 11
+                3, 3, 3, 3, 3, 3, 3, 3, 0, 0, 0,
+                0, 7, 7, 7, 7, 7, 7, 7, 7,  // 10
+                3, 3, 3, 3, 3, 3, 3, 4, 4, 0, 0,
+                6, 6, 7, 7, 7, 7, 7, 7, 7,  // 9
+                3, 3, 3, 3, 3, 3, 4, 4, 4, 5, 5,
+                6, 6, 6, 7, 7, 7, 7, 7, 7,  // 8
+                0, 3, 3, 3, 4, 4, 4, 4, 4, 5, 5,
+                6, 6, 6, 6, 6, 7, 7, 7, 0,  // 7
+                0, 3, 4, 4, 4, 4, 4, 4, 5, 5, 5,
+                5, 6, 6, 6, 6, 6, 6, 7, 0,  // 6
+                0, 4, 4, 4, 4, 4, 4, 4, 5, 5, 5,
+                5, 6, 6, 6, 6, 6, 6, 6, 0,  // 5
+                0, 0, 4, 4, 4, 4, 4, 4, 5, 5, 5,
+                5, 6, 6, 6, 6, 6, 6, 0, 0,  // 4
+                0, 0, 0, 4, 4, 4, 4, 5, 5, 5, 5,
+                5, 5, 6, 6, 6, 6, 0, 0, 0,  // 3
+                0, 0, 0, 0, 4, 4, 4, 5, 5, 5, 5,
+                5, 5, 6, 6, 6, 0, 0, 0, 0,  // 2
+                0, 0, 0, 0, 0, 0, 0, 5, 5, 5, 5,
+                5, 5, 0, 0, 0, 0, 0, 0, 0  // 1
+                // 1  2  3  4  5  6  7  8  9 10   11 12 13 14 15 16 17 18 19 20
+              };
+              
+              int iym, ixm, il, ic, ii;
+              iym = 20;
+              ixm = 20;
+              int iX_ = iX;
+              int iY_ = iY;
+              il = iym - iY_;
```

Comment:
just wondering - wouldn't it be simpler to make `idx_` a two-dimensional array, swap the order of the "rows" to avoid `20 - iY_`, and let  the compiler do the math ?

Reply:
same comment as before about a new PR with cpu and gpu improvements in parallel

https://github.com/cms-sw/cmssw/blob/2c91040e71a5447c0e161e80ce914260713b8317/CalibCalorimetry/EcalLaserAnalyzer/src/MEEEGeom.cc#L86-L93

(for reference about where to fix on cpu)